src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -161,6 +161,20 @@ public virtual IEnumerable<PackageResult> List(ChocolateyConfiguration config)
                 }
             }
 
+            if ((config.ListCommand.ApprovedOnly || config.ListCommand.DownloadCacheAvailable || config.ListCommand.NotBroken) && config.RegularOutput)
+            {
+                // This warning has been added here, to provide context to the user, and a follow up issues
+                // has been added here: https://github.com/chocolatey/choco/issues/3139 to address the actual
+                // issue that causes this warning to be required.
+                this.Log().Warn(@"
+Starting vith Chocolatey CLI v2.0.0, changes have been made to the
+`choco search` command which means that filtering of packages using the
+--approved-only, --download-cache, and --not-broken options are not
+performed within Chocolatey CLI, rather than on the repository that is
+being queried. As a result, it is possible that incomplete pages of
+packages are returned from a command that uses these options.");

```

Comment:
```suggestion
Starting vith Chocolatey CLI v2.0.0, changes have been made to the
`choco search` command which means that filtering of packages using the
--approved-only, --download-cache, and --not-broken options are not
performed within Chocolatey CLI. These are instead performed on the
repository that is being queried. As a result, it is possible that incomplete
package lists are returned from a command that uses these options.");

```

Reply:
Does this not just affect the Chocolatey Community Repository? If so, we should say 'performed by the Chocolatey Community Repository' instead of 'performed on the repository that is being queried'.

Reply:
This is a great question.  Chocolatey Community Repository is the only repository server that I know of that returns these properties for a package, and there is no real reason for another repository to return them, unless they have implemented there own package moderation system, and caching mechanism.

I think it is a fairly safe bet that they only impact the Chocolatey Community Repository, so I will get this section updated.